enhancements/helm3/console.md

```diff
@@ -102,10 +101,91 @@ spec:
+
+---
+
+Hence, it would be good to define the configuration as the top-level resource detaching API from the implementation. Cluster admin would easily discover the new resource and manage it either through UI. 
 
 ![Helm Cluster Configuration](assets/openshift-administration-cluster-settings.png)
 
+or via CLI:
+
+```shell
+$ oc get helmchartrepository
+NAME          AGE
+cluster       3h30m
+```
+
+In a case of multiple chart repositories, the console should be modified so that either allow:
+* editing all chart repository instances at once through aggregated YAML document
+* add/edit/removal of individual instances
+
+
+Adding an additional chart repositories via CLI follows the usual k8s pattern:
+
+```shell
+$ cat <<EOF | oc apply -f -
+apiVersion: helm.kubernetes.io/v1
+kind: HelmChartRepository
+metadata:
+  name: stable
+spec:
+  url: https://kubernetes-charts.storage.googleapis.com
+
+  displayName: Public Helm stable charts
+
+  description: Public Helm stable charts hosted on HelmHub
+---
+apiVersion: helm.kubernetes.io/v1
+kind: HelmChartRepository
+metadata:
+  name: incubator
+spec:
+  url: https://kubernetes-charts-incubator.storage.googleapis.com
+
+  displayName: Public Helm charts in incubator state
+EOF 
+
+$ oc get helmchartrepository
+NAME          AGE
+cluster       3h30m
+stable        1m
+incubator     1m
+```
+
+Chart repository proxy will use all configured chart repositories and deliver to the UI an aggregated index file. If needed by some future usecase, UI would be able read helm chart repository configuration and perhaps even talk to the individual chart repositories directly.
+
+Managing Helm releases via CLI could also benefit from the in-cluster stored chart repository configurations. If
+`HelmChartRepository` CRD includes all the fields from [Helm Repository Entry struct](https://github.com/helm/helm/blob/master/pkg/repo/chartrepo.go#L42), then the repositories declared in the cluster could be automatically used, without the need to issue `helm repo add` locally. These repositories will be shared across all developers accessing the cluster.
+
+Given the above example, the interaction could look like this:
+
+```shell
+$ oc get helmchartrepository
+NAME          AGE
+cluster       3h30m
+stable        1m
+incubator     1m
+
+# repositories configured in the cluster are implicitelly available for CLI as well
+$ helm repo list
+NAME     	URL
+cluster   http://my.chart-repo.org/stable
+stable   	https://kubernetes-charts.storage.googleapis.com           
+incubator	https://kubernetes-charts-incubator.storage.googleapis.com
+
+# install a chart from stable repo defined in the cluster
+$ helm install mysql stable/mysql
+```
+
+Currently Helm manages chart repository configurations in the local file only and the previously mentioned integration would require a contribution to the upstream. 
```

Comment:
I would prefer we add something about the behaviour we would envision --

-----

#### Relationship with the Helm CLI

* Currently the Helm CLI manages chart repository configurations in the local filesystem configuration directory only.
* In a separate upstream proposal, we shall propose making the CLI cluster-aware with respect to the chart repositories. In other words, `helm repo list` would not only list local file system configuration, it would also show us chart repositories represented as kubernetes resources in the cluster.